pxtcompiler/emitter/service.ts

```diff
@@ -528,6 +530,16 @@ namespace ts.pxtc {
                                 source = foundSrc;
                             }
                             si.attributes = parseCommentString(source);
+
+                            if (existing.kind === SymbolKind.Module) {
+                                // Copy the existing package array to the new symbol info
+                                si.pkgs = existing.pkgs || []
```

Comment:
don't pre-allocate pkgs; only do it when necessary since most of the time it's not used

Reply:
At this point we already have two colliding namespace symbols, we've also already either combined or picked one of their sources to keep. So when we set `pkgs` at this point, it's because we need to.